src/middlewared/middlewared/alert/source/memory_errors.py

```diff
@@ -0,0 +1,42 @@
+from collections import defaultdict
+from datetime import timedelta
+
+from middlewared.alert.base import AlertSource, IntervalSchedule
+
+
+class MemoryErrorsAlertSource(AlertSource):
+    schedule = IntervalSchedule(timedelta(hours=24))
+    products = ('SCALE_ENTERPRISE',)
+
+    async def check(self):
+        errors = defaultdict(dict)
+        alert = False
+        for mem_ctrl, info in (await self.middleware.call('hardware.memory.error_info')).items():
+            for key in ('uncorrected_errors', 'uncorrected_errors_with_no_dimm_info'):
+                if (value := info[key]) is not None and value > 0:
+                    # top-level memory controller
+                    errors[mem_ctrl].update({key: value})
+                    alert = True
+
+            for dimm_key in filter(lambda x: x.startswith(('dimm', 'rank')), info):
+                if (value := info[dimm_key]['uncorrected_errors']) is not None and value > 0:
+                    # the specific dimm
+                    errors[mem_ctrl].update({dimm_key: {'uncorrected_errors': value}})
+                    alert = True
+
+        if alert:
+            serial = await self.middleware.call('system.dmidecode_info')['system-serial-number']
+            job = await self.middleware.call('support.new_ticket', {
```

Comment:
If you don't want this method to emit an alert that is visible in the alerts UI section, there is no reason to make this an alert source, it can be just a periodically ran job.

However, if you add an alert class and set it with `proactive_support = True`, a support ticket will be created automatically (moreover, it won't be creating new support tickets every 24 hours)